File: 2582-pass-the-pillow/main.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        
        position, direction = 1, 1

        for _ in range(time):
            position += direction
            if position == n:
                direction = -1
            elif position == 1:
                direction = 1
        
        return position

File: 2582-pass-the-pillow/main.go
func passThePillow(n int, time int) int {
    
    position, direction := 1, 1

    for range time {
        position += direction
        if position == n {
            direction = -1
        }else if position == 1 {
            direction = 1
        } 
    }

    return position
}
